ipld/write.go

```diff
@@ -13,23 +13,16 @@ import (
 	"github.com/tendermint/tendermint/pkg/wrapper"
 )
 
-// BatchSize defines an amount of IPLD Nodes to be buffered and written at once.
-// This configuration is very database backend specific and the current default(128) is not optimized for the version of
-// Badger we use. We set it to one to avoid test flakiness, as some tests may read for data that was not written yet.
-// TODO(@Wondertan): Find out the perfect value for Badger(e.g. ask PL folks) or migrate to go-car IPLD
-//  storage(preferred).
-const BatchSize = 1
-
 // PutData posts erasured block data to IPFS using the provided ipld.NodeAdder.
 func PutData(ctx context.Context, shares [][]byte, adder ipld.NodeAdder) (*rsmt2d.ExtendedDataSquare, error) {
 	if len(shares) == 0 {
 		return nil, fmt.Errorf("empty data") // empty block is not an empty Data
 	}
-	// create nmt adder wrapping batch adder
-	batchAdder := NewNmtNodeAdder(ctx, ipld.NewBatch(ctx, adder, ipld.MaxSizeBatchOption(BatchSize)))
+	squareSize := int(math.Sqrt(float64(len(shares))))
+	// create nmt adder wrapping batch adder with calculated size
+	batchAdder := NewNmtNodeAdder(ctx, ipld.NewBatch(ctx, adder, ipld.MaxSizeBatchOption(batchSize(squareSize))))
```

Comment:
> `squareSize := int(math.Sqrt(float64(len(shares))))`

These are only the orig. square shares, right? But I thought `batchSize` was supposed to return the number of all nodes in the extended square? 
```suggestion
	squareSize := int(math.Sqrt(float64(len(shares))))
	// create nmt adder wrapping batch adder with calculated size
	bs := batchSize(squareSize * 2)
	batchAdder := NewNmtNodeAdder(ctx, ipld.NewBatch(ctx, adder, ipld.MaxSizeBatchOption(bs)))
```